### ISSUE TITLE
Unicode icons in textboxes are not rendered correctly

### ISSUE THREAD

**Issue by scullhead** [greenshot/greenshot#283]
Greenshot V1.2.10 Build 6 (64 Bit)
Windows 10 20H2

- It is possible to insert unicode emojis and icons into text boxes (e.G.: 😁✔ 🠆, ...) by copy & paste or by Pressing the "Win + . " Keys

- The unicode icons are shown corectly when the text box is in "edit mode":<br />![Unicode_Icons_Text_Edit_Mode](https://user-images.githubusercontent.com/10958573/108708622-f75ea180-7511-11eb-8e4d-07e4927b26c1.png)

- When you leave the edit mode, some icons are shown (but differently than in edit mode ( e.g. the check mark ✔) and some are not shown at all (just rectangles):<br />![Unicode_Icons_Preview_Mode](https://user-images.githubusercontent.com/10958573/108708778-2c6af400-7512-11eb-991e-b2ae8e34bb74.png)

**Comment by Lakritzator** [comment]
I've added the fix to the most recent 1.3 build (1.3.151 or later), which is not officially released yet. 
You can find it here: https://github.com/greenshot/greenshot/releases

**Comment by Lakritzator** [comment]
I actually have to roll-back my changes, the fix caused more harm than good.
It seems that the routine which allows me to use unicode, has other and much worse issues.

**Comment by Lakritzator** [comment]
Just rolled back the change, yeah it made Unicode icons visible, but it caused a lot of issues.
I currently don't know how to solve it, without having more other issues.

Marking it as "known issue"

**Comment by scullhead** [comment]
Ok, thanks for the info. Maybe it will be fixed another time. Currently it is more like a "nice to have" feature.

**Comment by Lakritzator** [comment]
I fully agree this should be fixed, but if Windows doesn't offer a good API for the technology we are currently using for the editor (Windows Forms), it has to wait until we finally move to use WPF (or something else).

As a workaround, you should be able to drag & drop images from websites on the drawing canvas, this could be emoticons. Doesn't work with EVERY website, but a lot should work.

**Comment by jairbubbles** [comment]
Hi @Lakritzator, have you considered using `TextRenderer` only when the text contains emojis / icons?

As an alternative, what would you think of adding a new object type "emoji" so that normal text blocks wouldn't be affected? It would help annotate and give impressions on a screenshot.

**Comment by jairbubbles** [comment]
I dug a bit more and it looks like using a font that embeds directly the unicode emoji fixes the issue.
I'm proposing to use "Segoe UI Emoji" by default, let me know what you think! => #357

**Comment by Lakritzator** [comment]
I actually thought about your initial suggestion, but found out that emojis get added to the standard and I would need to keep track of them. I did find some regexp to check for them, which might have make it possible BUT:
it will not happen that oft, that one only has 1 emoji... it might be a small fix but not the correct one.

I was going to suggest that it needs to wait for the long due rewrite of the editor...

But your second suggestion is interesting, I had a quick look and it's cool.
Is Segeo UI Emoji always installed, or is this only available when the user does something?
Maybe it's dependend on the Windows version (is it available in 7 & 8?), 1.3 would still work on Windows 7!

I'm a bit hassitent to change the default font, but it's certainly a very good workaround.

**Comment by jairbubbles** [comment]
> Is Segeo UI Emoji always installed, or is this only available when the user does something?

Looks like it's only Windows 10+ (see https://docs.microsoft.com/fr-fr/typography/font-list/segoe-ui-emoji).  Anyway in my PR I test if it's available before setting it.

> I was going to suggest that it needs to wait for the long due rewrite of the editor...

WPF is dying slowly too and does not support colour emoji by default. Have you checked [Avalonia](https://avaloniaui.net/)? I just tested https://icons8.com/lunacy which is a drawing application powered by Avalonia. I'm pretty impressed by the quality and it supports Emojis 😏

![image](https://user-images.githubusercontent.com/15875066/148653119-5a9e5e85-66dd-4b84-8f8d-a8f076c839e8.png)

(https://dev.to/avalonia/case-study-lunacy-2fg9)

**Comment by Lakritzator** [comment]
Yes, I have been playing around with Avalonia for the last months and I love it, I have been considering this in favor of WPF for Greenshot 2.
Although I do not see WPF dying short term, but the advantage of using Avalonia is that the way for a Greenshot on Linux is open.

I do see some issues though, but this is not directly related to what you see, but the stuff in the background.
Avalonia bases upon Skia, so it uses Skia Bitmaps. I was thinking of using ImageSharp for the bitmap processing, and I do not yet know of a good solution there. But I guess time will solve that.

**Comment by jairbubbles** [comment]
> Yes, I have been playing around with Avalonia for the last months and I love it, I have been considering this in favor of WPF 

I clearly wouldn't start an app in WPF in 2022. I've been playing with Blazor Desktop lately and it's worth considering too IMO. But it means no more XAML 😅

> for Greenshot 2.

😍
